keps/sig-auth/2907-secrets-store-csi-driver/README.md

```diff
@@ -0,0 +1,368 @@
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+- non-zero `total_node_publish` metrics indicate the CSI driver is used by the workloads.
+- `total_sync_k8s_secret` metrics indicate the optional Sync as Kubernetes secret feature is used by the workloads.
+- `total_rotation_reconcile` metrics indicate the optional rotation reconciliation feature is used by the workloads.
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- `total_node_publish_error`
+  - any rising count of this metric indicates a problem with mounting the volume for pod.
+- `total_node_publish_error`
+  - any rising count of this metric indicates a problem with unmounting the volume for pod.
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [x] Metrics
+  - Metric name: `total_node_publish`
+  - Components exposing the metric: `secrets-store-csi-driver`
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+### Dependencies
+
+- [Kubernetes Container Storage Interface](https://github.com/kubernetes/community/blob/98b3d97d2e7f91bb62b8e88710c29c1675efb689/contributors/design-proposals/storage/container-storage-interface.md)
+- [KEP 596](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/596-csi-inline-volumes)
+- Supports windows containers (Kubernetes version v1.18+)
+- [KEP 1855: Service Account Token for CSI Driver](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1855-csi-driver-service-account-token)
+
+###### Does this feature depend on any specific services running in the cluster?
+
```

Comment:
May mention k8s version required?

Reply:
Added the minimum required Kubernetes version and called out the kubelet dependency for volume mount.